doc/voq/voq_hld.md

```diff
@@ -0,0 +1,813 @@
+ ![](../../images/voq_hld/network-to-cpu-packet-flows.png)
+
+## 1.6 CLI requirements
+User should be able to display/retrieve the following VOQ information
+- system-port id for the system ports
+- Switch-id
+- max-cores
+- Neighbor Encap index
+
+## 1.7 Scalability requirements
+
+###### Table 2: VOQ scalability
+| VOQ component            | Expected value              |
+|--------------------------|-----------------------------|
+| VOQ Switches             | 128                         |
+| System Ports             | 4K                          |
+| VOQ Neighbors            | 16k                         |
+
+The initial scaling should at least support the requirements of a "large" modular chassis deploying ports based routing interfaces. Virtual interfaces (VLAN based routing interfaces) and directly attached hosts are still supported, but achieving higher scaling for those deployments is not a goal of the initial implementation.
+
+## 1.8 Warm Restart requirements
+At this time warm restart capability is not factored into the design. This shall be revisited as a Phase #2 enhancement
+
+# 2 Modules Design
+### 2.1 System Port Configuration on Sonic Instance
+The system ports are configured on the Line Card. This information is then populated in the "System Port Table" in the "Config DB". On each sonic instance - the "OrchAgent" daemon subscribes to this information and recevies it. 
+
+ ![](../../images/voq_hld/control-card-system-port-config-flow.png)
+
+## 2.2 Config DB
+
+### 2.2.1 DEVICE_METADATA
+ The **existing** DEVICE_METADATA table is enhanced to add new entry to have VOQ related parameters
+ 
+```
+DEVICE_METADATA: {
+   "voqsystem": {
+    "switch_type": {{switch_type}}
+    "switch_id": {{switch_id}}
+    "max_cores" : {{max_cores}}
+   }
+}
+```
+
+VOQ_DB will be specified using database_config.json.j2 and it will generated in control card and linecard differently. Control card will be the server and line card will be running clients only. DBConnectors will use this config to connect correct VOQ_DB.
+
+sonic-buildimage/dockers/docker-database/database_config.json.j2
+
+```
+    "INSTANCES": {
+        "redis":{
+            "hostname" : "{{HOST_IP}}",
+            "port" : 6379,
+            "unix_socket_path" : "/var/run/redis{{NAMESPACE_ID}}/redis.sock",
+            "persistence_for_warm_boot" : "yes"
+        },
+{%- if sonic_asic_platform == "voq" %}
+        "voq-redis":{
+            "hostname" : "{{HOST_IP}}",
+            "port" : 6379,
+            "unix_socket_path" : "/var/run/redis{{NAMESPACE_ID}}/redis.sock",
+            "persistence_for_warm_boot" : "no",
+{%- if sonic_asic_platform_cardtype == "linecard" %}
+            "run_server": "no"
+{%- endif %}
+        }
+{%- endif %}
+    },
+    "DATABASES" : {
+        "APPL_DB" : {
+            "id" : 0,
+            "separator": ":",
+            "instance" : "redis"
+        },
+        ...
+        "VOQ_DB" : {
```

Comment:
As mentioned above, this should be GLOBAL_APP_DB

Reply:
Agreed. We'll update the hld